### PR TITLE
Sub-daily granularity test additions

### PR DESCRIPTION
**Pull request by courtneyholcomb**
This PR handles some PR feedback related to testing for sub-daily granularity that I wanted to add at the top of the stack to avoid too many issues with rebasing. Updates:
- Changes the dates of the test time spine data to span across multiple days
- Adds tests related to the interactions between sub-daily granularity and metric default granularity

### CODE REVIEW

File: tests_metricflow/fixtures/source_table_snapshots/time_spine_table/mf_time_spine_microsecond.yaml [hunk 1]
```diff
@@ -15,23 +15,23 @@ table_snapshot:
...
+    - ["2020-01-02 00:00:00.000012"]
+    - ["2020-01-02 00:00:00.000013"]
+    - ["2020-01-02 00:00:00.000014"]
+    - ["2020-01-02 00:00:00.000015"]
+    - ["2020-01-02 00:00:00.000016"]
+    - ["2020-01-02 00:00:00.000017"]
+    - ["2020-01-02 00:00:00.000018"]
+    - ["2020-01-02 00:00:00.000019"]
+    - ["2020-01-02 00:00:00.000020"]
+    - ["2020-01-03 00:00:00.000021"]
```
Comment (tlento) [inline tests_metricflow/fixtures/source_table_snapshots/time_spine_table/mf_time_spine_microsecond.yaml:28]: Argh I just realized these will no longer be contiguous, but I think that's fine. I guess we could do a midnight span but eh....

File: metricflow-semantics/metricflow_semantics/test_helpers/semantic_manifest_yamls/simple_manifest/semantic_models/user_sm_source.yaml [hunk 2]
```diff
@@ -106,3 +106,20 @@ metric:
...
+  name: simple_subdaily_metric_default_day
+  description: simple metric with sub-daily agg time dim that joins to time spine
+  type: simple
+  type_params:
+    measure:
+      name: archived_users
+---
+metric:
+  name: simple_subdaily_metric_default_hour
+  description: simple metric with sub-daily agg time dim that joins to time spine
```
Comment (tlento) [inline metricflow-semantics/metricflow_semantics/test_helpers/semantic_manifest_yamls/simple_manifest/semantic_models/user_sm_source.yaml:120]: nit: these descriptions don't seem right.

File: tests_metricflow/integration/test_cases/itest_metrics.yaml [hunk 3]
```diff
@@ -2330,3 +2330,29 @@ integration_test:
...
+  check_query: |
+    SELECT
+      {{ render_date_trunc("archived_at", TimeGranularity.DAY) }} AS metric_time__day
+      , SUM(1) AS simple_subdaily_metric_default_day
+    FROM {{ source_schema }}.dim_users
+    GROUP BY {{ render_date_trunc("archived_at", TimeGranularity.DAY) }}
+---
+integration_test:
+  name: subdaily_metric_default_granularity
+  description: test that a subdaily metric without explicit default granularity defaults to DAY
```
Comment (tlento) [inline tests_metricflow/integration/test_cases/itest_metrics.yaml:2349]: nit:
```suggestion
  description: test that a subdaily metric with an explicit default granularity uses its specified default
```

**Review by tlento — APPROVED** [review]

File: metricflow-semantics/metricflow_semantics/test_helpers/semantic_manifest_yamls/simple_manifest/semantic_models/user_sm_source.yaml (hunk 2 above)
Comment (courtneyholcomb) [reply metricflow-semantics/metricflow_semantics/test_helpers/semantic_manifest_yamls/simple_manifest/semantic_models/user_sm_source.yaml:120]: whoops that was a copy paste for sure.

**Pull request merged**